### PR TITLE
Fix HMCECS for likelihoods with multiple plates

### PR DESCRIPTION
**Pull request by fehiepsi**
Currently, when the likelihood has multiple plate statements, HMCECS will raise errors. This PR fixes that bug.

### CODE REVIEW

File: numpyro/infer/hmc_gibbs.py [hunk 1]
```diff
@@ -787,7 +789,7 @@ def log_likelihood(params_flat, subsample_indices=None):
                             log_lik[frame.name] += _sum_all_except_at_dim(
                                 site["fn"].log_prob(site["value"]), frame.dim
                             )
-                        else:
+                        elif frame.name in subsample_indices:
```
Comment (fehiepsi) [inline numpyro/infer/hmc_gibbs.py:792]: This is where the bug happens. We only want to collect likelihoods at subsample dimensions.

Comment (OlaRonning) [inline numpyro/infer/hmc_gibbs.py:792]: That makes sense. 
I think you could merge the conditionals with a `defaultdict`. Something like
``` python
log_lik = defaultdict(float)
for site in tr.values():
  if site["type"] == "sample" and site["is_observed"]:
    for frame in site["cond_indep_stack"]:
      if frame.name in log_lik or frame.name in subsample_indices:
        log_lik[frame.name] += _sum_all_except_at_dim(
          site["fn"].log_prob(site["value"]), frame.dim)
```

**Review by OlaRonning — APPROVED** [review]
Thanks, @fehiepsi. The PR looks good. Have a minor comment.

File: numpyro/infer/hmc_gibbs.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/infer/hmc_gibbs.py:792]: I think this works but we need to maintain the type of the returned values. If we use defaultdict here, we will need to use defaultdict in other places like in gibbs_update function below.

**Pull request merged**